### DISCUSSION TITLE
Autosave changes

### DISCUSSION THREAD

**Discussion by mrc-eu** [JetClient/jet-client-support#70] [Ideas]
Why isn't there an option to auto-save the changes?

**Comment by AntonShuvaev** [comment]
In JetClient, I aimed to combine both automatic and manual saving. All changes to requests and folders are automatically saved to the plugin's state file, ensuring you won’t lose any changes when you close your tabs. However, saving and restoring changes to and from sync files must be done manually.

I initially chose not to implement auto-sync for a couple of reasons:
1) Not every user uses a VCS for JetClient sync files; some prefer to maintain a "correct" version of their request collections on their filesystem, allowing them to save and restore only when needed.
2) To prevent temporary changes that you don't want to commit from conflicting with updates or branch switches.

However, I recognize that manually saving and restoring can be redundant for users who do use a VCS.
I will definitely consider adding an option for auto-sync in the near future. Thanks for bringing this up!

**Reply by mrc-eu** [reply]
Thank you. For VCS users, like me, that would be extremely helpful.

**Comment by AntonShuvaev** [comment]
The auto-sync feature has been added in version 2024.2.3
